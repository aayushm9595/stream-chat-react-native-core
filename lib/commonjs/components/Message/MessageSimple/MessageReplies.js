var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.MessageReplies = void 0;

var _taggedTemplateLiteralLoose2 = _interopRequireDefault(require("@babel/runtime/helpers/taggedTemplateLiteralLoose"));

var _react = _interopRequireDefault(require("react"));

var _TranslationContext = require("../../../contexts/translationContext/TranslationContext");

var _styledComponents = require("../../../styles/styledComponents");

var _jsxFileName = "/Users/vishalnarkhede/projects/projects/stream-chat-react-native/src/components/Message/MessageSimple/MessageReplies.tsx";

function _templateObject3() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  color: ", ";\n  font-size: 12px;\n  font-weight: 700;\n  ", "\n"]);

  _templateObject3 = function _templateObject3() {
    return data;
  };

  return data;
}

function _templateObject2() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  transform: ", ";\n  ", "\n"]);

  _templateObject2 = function _templateObject2() {
    return data;
  };

  return data;
}

function _templateObject() {
  var data = (0, _taggedTemplateLiteralLoose2["default"])(["\n  align-items: center;\n  flex-direction: row;\n  padding: 5px;\n  ", "\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

var iconPath = require("../../../images/icons/icon_path.png");

var Container = _styledComponents.styled.TouchableOpacity(_templateObject(), function (_ref) {
  var theme = _ref.theme;
  return theme.message.replies.container.css;
});

var MessageRepliesImage = _styledComponents.styled.Image(_templateObject2(), function (_ref2) {
  var alignment = _ref2.alignment;
  return alignment === 'left' ? 'rotateY(0deg)' : 'rotateY(180deg)';
}, function (_ref3) {
  var theme = _ref3.theme;
  return theme.message.replies.image.css;
});

var MessageRepliesText = _styledComponents.styled.Text(_templateObject3(), function (_ref4) {
  var theme = _ref4.theme;
  return theme.colors.primary;
}, function (_ref5) {
  var theme = _ref5.theme;
  return theme.message.replies.messageRepliesText.css;
});

var MessageReplies = function MessageReplies(props) {
  var alignment = props.alignment,
      isThreadList = props.isThreadList,
      message = props.message,
      openThread = props.openThread;

  var _useTranslationContex = (0, _TranslationContext.useTranslationContext)(),
      t = _useTranslationContex.t;

  if (isThreadList || !message.reply_count) return null;
  return _react["default"].createElement(Container, {
    onPress: openThread,
    testID: "message-replies",
    __source: {
      fileName: _jsxFileName,
      lineNumber: 86,
      columnNumber: 5
    }
  }, alignment === 'left' && _react["default"].createElement(MessageRepliesImage, {
    alignment: alignment,
    source: iconPath,
    testID: "message-replies-left",
    __source: {
      fileName: _jsxFileName,
      lineNumber: 88,
      columnNumber: 9
    }
  }), _react["default"].createElement(MessageRepliesText, {
    __source: {
      fileName: _jsxFileName,
      lineNumber: 94,
      columnNumber: 7
    }
  }, message.reply_count === 1 ? t('1 reply') : t('{{ replyCount }} replies', {
    replyCount: message.reply_count
  })), alignment === 'right' && _react["default"].createElement(MessageRepliesImage, {
    alignment: alignment,
    source: iconPath,
    testID: "message-replies-right",
    __source: {
      fileName: _jsxFileName,
      lineNumber: 100,
      columnNumber: 9
    }
  }));
};

exports.MessageReplies = MessageReplies;
//# sourceMappingURL=MessageReplies.js.map