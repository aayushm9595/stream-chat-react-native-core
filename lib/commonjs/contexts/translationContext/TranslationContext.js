var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

var _interopRequireWildcard = require("@babel/runtime/helpers/interopRequireWildcard");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.withTranslationContext = exports.useTranslationContext = exports.TranslationProvider = exports.TranslationContext = exports.isDayOrMoment = void 0;

var _extends2 = _interopRequireDefault(require("@babel/runtime/helpers/extends"));

var _react = _interopRequireWildcard(require("react"));

var _dayjs = _interopRequireDefault(require("dayjs"));

var _getDisplayName = require("../utils/getDisplayName");

var _jsxFileName = "/Users/vishalnarkhede/projects/projects/stream-chat-react-native/src/contexts/translationContext/TranslationContext.tsx";

var isDayOrMoment = function isDayOrMoment(output) {
  return output.isSame != null;
};

exports.isDayOrMoment = isDayOrMoment;

var TranslationContext = _react["default"].createContext({
  t: function t(key) {
    return key;
  },
  tDateTimeParser: function tDateTimeParser(input) {
    return (0, _dayjs["default"])(input);
  }
});

exports.TranslationContext = TranslationContext;

var TranslationProvider = function TranslationProvider(_ref) {
  var children = _ref.children,
      value = _ref.value;
  return _react["default"].createElement(TranslationContext.Provider, {
    value: value,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 42,
      columnNumber: 3
    }
  }, children);
};

exports.TranslationProvider = TranslationProvider;

var useTranslationContext = function useTranslationContext() {
  return (0, _react.useContext)(TranslationContext);
};

exports.useTranslationContext = useTranslationContext;

var withTranslationContext = function withTranslationContext(Component) {
  var WithTranslationContextComponent = function WithTranslationContextComponent(props) {
    var translationContext = useTranslationContext();
    return _react["default"].createElement(Component, (0, _extends2["default"])({}, props, translationContext, {
      __source: {
        fileName: _jsxFileName,
        lineNumber: 57,
        columnNumber: 12
      }
    }));
  };

  WithTranslationContextComponent.displayName = "WithTranslationContext" + (0, _getDisplayName.getDisplayName)(Component);
  return WithTranslationContextComponent;
};

exports.withTranslationContext = withTranslationContext;
//# sourceMappingURL=TranslationContext.js.map