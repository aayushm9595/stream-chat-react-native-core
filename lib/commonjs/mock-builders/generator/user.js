var _interopRequireDefault = require("@babel/runtime/helpers/interopRequireDefault");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.generateStaticUser = exports.generateUser = void 0;

var _defineProperty2 = _interopRequireDefault(require("@babel/runtime/helpers/defineProperty"));

var _uuid = require("uuid");

function ownKeys(object, enumerableOnly) { var keys = Object.keys(object); if (Object.getOwnPropertySymbols) { var symbols = Object.getOwnPropertySymbols(object); if (enumerableOnly) symbols = symbols.filter(function (sym) { return Object.getOwnPropertyDescriptor(object, sym).enumerable; }); keys.push.apply(keys, symbols); } return keys; }

function _objectSpread(target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i] != null ? arguments[i] : {}; if (i % 2) { ownKeys(Object(source), true).forEach(function (key) { (0, _defineProperty2["default"])(target, key, source[key]); }); } else if (Object.getOwnPropertyDescriptors) { Object.defineProperties(target, Object.getOwnPropertyDescriptors(source)); } else { ownKeys(Object(source)).forEach(function (key) { Object.defineProperty(target, key, Object.getOwnPropertyDescriptor(source, key)); }); } } return target; }

var generateUser = function generateUser() {
  var options = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : {};
  return _objectSpread({
    banned: false,
    created_at: '2020-04-27T13:39:49.331742Z',
    id: (0, _uuid.v4)(),
    image: (0, _uuid.v4)(),
    name: (0, _uuid.v4)(),
    online: false,
    role: 'user',
    updated_at: '2020-04-27T13:39:49.332087Z'
  }, options);
};

exports.generateUser = generateUser;

var generateStaticUser = function generateStaticUser(userNumber) {
  switch (userNumber) {
    case 0:
      return generateUser({
        id: 'DannyBoi1',
        image: 'https://images-na.ssl-images-amazon.com/images/I/51t29lLkg8L._AC_SL1000_.jpg',
        name: 'Dan'
      });

    case 1:
      return generateUser({
        id: 'GrantyBoi2',
        image: 'https://i.imgur.com/SLx06PP.png',
        name: 'Grant'
      });

    case 2:
      return generateUser({
        id: 'ViriBoi2',
        image: 'https://i.imgur.com/iNaC3K7.jpg',
        name: 'Vir'
      });

    default:
      return generateUser({
        id: 'NeiliBoi3',
        image: 'https://i.imgur.com/T68W8nR_d.webp?maxwidth=728&fidelity=grand',
        name: 'Neil'
      });
  }
};

exports.generateStaticUser = generateStaticUser;
//# sourceMappingURL=user.js.map