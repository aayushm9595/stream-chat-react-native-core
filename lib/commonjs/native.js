Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.registerNativeHandlers = exports.pickDocument = exports.pickImage = exports.NetInfo = void 0;

var fail = function fail() {
  throw Error('Native handler was not registered, you should import stream-chat-expo or stream-chat-react-native');
};

var NetInfo = {
  addEventListener: fail,
  fetch: fail
};
exports.NetInfo = NetInfo;
var pickImage = fail;
exports.pickImage = pickImage;
var pickDocument = fail;
exports.pickDocument = pickDocument;

var registerNativeHandlers = function registerNativeHandlers(handlers) {
  if (handlers.NetInfo) {
    exports.NetInfo = NetInfo = handlers.NetInfo;
  }

  if (handlers.pickImage) {
    exports.pickImage = pickImage = handlers.pickImage;
  }

  if (handlers.pickDocument) {
    exports.pickDocument = pickDocument = handlers.pickDocument;
  }
};

exports.registerNativeHandlers = registerNativeHandlers;
//# sourceMappingURL=native.js.map