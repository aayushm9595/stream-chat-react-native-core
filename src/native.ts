import type { NetInfoSubscription } from '@react-native-community/netinfo';

const fail = () => {
  throw Error(
    'Native handler was not registered, you should import stream-chat-expo or stream-chat-react-native',
  );
};

type NetInfo = {
  addEventListener: (
    listener: (isConnected: boolean) => void,
  ) => NetInfoSubscription | never;
  fetch: (requestedInterface?: string | undefined) => Promise<boolean> | never;
};

export let NetInfo: NetInfo = {
  addEventListener: fail,
  fetch: fail,
};

type PickImage = ({
  compressImageQuality,
  maxNumberOfFiles,
}: {
  compressImageQuality?: number;
  maxNumberOfFiles?: number;
}) => Promise<{ cancelled: boolean; images?: { uri: string }[] }> | never;
export let pickImage: PickImage = fail;

type PickDocument = ({
  maxNumberOfFiles,
}: {
  maxNumberOfFiles?: number;
}) =>
  | Promise<{
      cancelled: boolean;
      docs?: {
        name: string;
        size?: number | string;
        type?: string;
        uri?: string;
      }[];
    }>
  | never;
export let pickDocument: PickDocument = fail;

type Handlers = {
  NetInfo?: NetInfo;
  pickDocument?: PickDocument;
  pickImage?: PickImage;
};
export const registerNativeHandlers = (handlers: Handlers) => {
  if (handlers.NetInfo) {
    NetInfo = handlers.NetInfo;
  }

  if (handlers.pickImage) {
    pickImage = handlers.pickImage;
  }

  if (handlers.pickDocument) {
    pickDocument = handlers.pickDocument;
  }
};
