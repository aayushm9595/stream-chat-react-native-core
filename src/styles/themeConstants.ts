import type { ImageStyle, StyleProp, TextStyle, ViewStyle } from 'react-native';
import type { CSSObject } from 'styled-components';

export const BASE_FONT_SIZE = 16;

export const Colors = {
  danger: '#EDD8DD',
  light: '#EBEBEB',
  primary: '#006cff',
  secondary: '#111',
  textDark: 'rgba(0,0,0,1)',
  textGrey: 'rgba(0,0,0,0.5)',
  textLight: 'white',
  transparent: 'transparent',
};

export type CSS = {
  css?: CSSObject;
};

export type MarkdownStyle = Partial<{
  blockQuoteBar: StyleProp<ViewStyle>;
  blockQuoteSection: StyleProp<ViewStyle>;
  blockQuoteSectionBar: StyleProp<ViewStyle>;
  blockQuoteText: StyleProp<ViewStyle>;
  codeBlock: StyleProp<TextStyle>;
  del: StyleProp<TextStyle>;
  em: StyleProp<TextStyle>;
  heading: StyleProp<TextStyle>;
  heading1: StyleProp<TextStyle>;
  heading2: StyleProp<TextStyle>;
  heading3: StyleProp<TextStyle>;
  heading4: StyleProp<TextStyle>;
  heading5: StyleProp<TextStyle>;
  heading6: StyleProp<TextStyle>;
  hr: StyleProp<ViewStyle>;
  image: StyleProp<ImageStyle>;
  inlineCode: StyleProp<TextStyle>;
  link: StyleProp<TextStyle>;
  listItem: StyleProp<ViewStyle>;
  listItemBulletType: string;
  listItemButton: StyleProp<TextStyle>;
  listItemNumber: StyleProp<TextStyle>;
  listItemText: StyleProp<TextStyle>;
  mailTo: StyleProp<TextStyle>;
  paragraph: StyleProp<ViewStyle>;
  strong: StyleProp<TextStyle>;
  table: StyleProp<ViewStyle>;
  tableHeader: StyleProp<ViewStyle>;
  tableHeaderCell: StyleProp<ViewStyle>;
  tableRow: StyleProp<ViewStyle>;
  tableRowCell: StyleProp<ViewStyle>;
  tableRowLast: StyleProp<ViewStyle>;
  text: StyleProp<TextStyle>;
  u: StyleProp<TextStyle>;
  video: StyleProp<ImageStyle>;
  view: StyleProp<ViewStyle>;
}>;

export type Theme = {
  avatar: {
    container: CSS;
    fallback: CSS;
    image: CSS;
    text: CSS;
  };
  channelListFooterLoadingIndicator: {
    container: CSS;
  };
  channelListHeaderErrorIndicator: {
    container: CSS;
    errorText: CSS;
  };
  channelListHeaderNetworkDownIndicator: {
    container: CSS;
    errorText: CSS;
  };
  channelPreview: {
    container: CSS;
    date: CSS;
    details: CSS;
    detailsTop: CSS;
    message: CSS & {
      color: string;
      fontWeight: string;
      unreadColor: string;
      unreadFontWeight: string;
    };
    title: CSS;
  };
  closeButton: {
    container: CSS;
  };
  colors: typeof Colors;
  iconBadge: {
    container: CSS;
    icon: CSS;
    iconInner: CSS;
    unreadCount: CSS;
  };
  iconSquare: {
    container: CSS;
  };
  loadingErrorIndicator: {
    container: CSS;
    errorText: CSS;
    retryText: CSS;
  };
  loadingIndicator: {
    container: CSS;
    loadingText: CSS;
  };
  message: {
    actions: {
      button: CSS & {
        defaultBackgroundColor: string;
        defaultBorderColor: string;
        primaryBackgroundColor: string;
        primaryBorderColor: string;
      };
      buttonText: CSS & {
        defaultColor: string;
        primaryColor: string;
      };
      container: CSS;
    };
    actionSheet: {
      buttonContainer: CSS;
      buttonText: CSS;
      cancelButtonContainer: CSS;
      cancelButtonText: CSS;
      titleContainer: CSS;
      titleText: CSS;
    };
    avatarWrapper: {
      container: CSS;
      spacer: CSS;
    };
    card: {
      container: CSS;
      cover: CSS;
      footer: CSS & {
        description: CSS;
        link: CSS;
        logo: CSS;
        title: CSS;
      };
    };
    container: CSS;
    content: {
      container: CSS & {
        borderRadiusL: number;
        borderRadiusS: number;
      };
      containerInner: CSS;
      deletedContainer: CSS;
      deletedText: CSS;
      errorContainer: CSS & {
        backgroundColor: string;
      };
      // Available options for styling text: https://github.com/CharlesMangwa/react-native-simple-markdown/tree/next#styles-1
      markdown: MarkdownStyle;
      metaContainer: CSS;
      metaText: CSS;
      textContainer: CSS & {
        borderRadiusL: number;
        borderRadiusS: number;
        leftBorderColor: string;
        leftBorderWidth: number;
        rightBorderColor: string;
        rightBorderWidth: number;
      };
    };
    file: {
      container: CSS;
      details: CSS;
      icon: CSS;
      size: CSS;
      title: CSS;
    };
    gallery: {
      doubleSize: number;
      galleryContainer: CSS;
      halfSize: number;
      header: {
        button: CSS;
        container: CSS;
      };
      imageContainer: CSS;
      single: CSS;
      size: number;
      width: number;
    };
    reactionList: {
      container: CSS;
      reactionCount: CSS;
    };
    reactionPicker: {
      column: CSS;
      container: CSS;
      containerView: CSS;
      emoji: CSS;
      reactionCount: CSS;
      text: CSS;
    };
    replies: {
      container: CSS;
      image: CSS;
      messageRepliesText: CSS;
    };
    status: {
      checkMark: CSS;
      deliveredCircle: CSS;
      deliveredContainer: CSS;
      readByContainer: CSS;
      readByCount: CSS;
      sendingContainer: CSS;
      sendingImage: CSS;
      spacer: CSS;
    };
  };
  messageInput: {
    actionSheet: {
      buttonContainer: CSS;
      buttonText: CSS;
      titleContainer: CSS;
      titleText: CSS;
    };
    attachButton: CSS;
    attachButtonIcon: CSS;
    container: CSS & {
      conditionalPadding: number;
    };
    editingBoxContainer: CSS;
    editingBoxHeader: CSS;
    editingBoxHeaderTitle: CSS;
    fileUploadPreview: {
      attachmentContainerView: CSS;
      attachmentView: CSS;
      container: CSS;
      dismiss: CSS;
      dismissImage: CSS;
      filenameText: CSS;
      itemContainer: CSS;
    };
    imageUploadPreview: {
      container: CSS;
      dismiss: CSS;
      dismissImage: CSS;
      itemContainer: CSS;
      upload: CSS;
    };
    inputBox: CSS;
    inputBoxContainer: CSS;
    sendButton: CSS;
    sendButtonIcon: CSS;
    suggestions: {
      command: {
        args: CSS;
        container: CSS;
        description: CSS;
        title: CSS;
        top: CSS;
      };
      container: CSS & {
        itemHeight: number;
        maxHeight: number;
      };
      item: CSS;
      mention: {
        container: CSS;
        name: CSS;
      };
      separator: CSS;
      title: CSS;
      wrapper: CSS;
    };
    uploadProgressIndicator: {
      container: CSS;
      overlay: CSS;
    };
  };
  messageList: {
    dateSeparator: {
      container: CSS;
      date: CSS;
      dateText: CSS;
      line: CSS;
    };
    errorNotification: CSS;
    errorNotificationText: CSS;
    eventIndicator: {
      date: CSS;
      memberUpdateContainer: CSS;
      memberUpdateText: CSS;
      memberUpdateTextContainer: CSS;
    };
    listContainer: CSS;
    messageNotification: {
      container: CSS;
      text: CSS;
    };
    messageSystem: {
      container: CSS;
      dateText: CSS;
      line: CSS;
      text: CSS;
      textContainer: CSS;
    };
    typingIndicatorContainer: CSS;
  };
  spinner: CSS;
  thread: {
    newThread: CSS & {
      text: CSS;
    };
  };
  typingIndicator: {
    container: CSS;
    text: CSS & {
      color: string;
      fontSize: number;
    };
  };
};

export const defaultTheme: Theme = {
  avatar: {
    container: {},
    fallback: {},
    image: {},
    text: {},
  },
  channelListFooterLoadingIndicator: {
    container: {},
  },
  channelListHeaderErrorIndicator: {
    container: {},
    errorText: {},
  },
  channelListHeaderNetworkDownIndicator: {
    container: {},
    errorText: {},
  },
  channelPreview: {
    container: {},
    date: {},
    details: {},
    detailsTop: {},
    message: {
      color: '#767676',
      fontWeight: 'normal',
      unreadColor: '#000',
      unreadFontWeight: 'bold',
    },
    title: {},
  },
  closeButton: {
    container: {},
  },
  colors: {
    ...Colors,
  },
  iconBadge: {
    container: {},
    icon: {},
    iconInner: {},
    unreadCount: {},
  },
  iconSquare: {
    container: {},
  },
  loadingErrorIndicator: {
    container: {},
    errorText: {},
    retryText: {},
  },
  loadingIndicator: {
    container: {},
    loadingText: {},
  },
  message: {
    actions: {
      button: {
        defaultBackgroundColor: 'white',
        defaultBorderColor: 'transparent',
        primaryBackgroundColor: Colors.primary,
        primaryBorderColor: Colors.light,
      },
      buttonText: {
        defaultColor: 'black',
        primaryColor: 'white',
      },
      container: {},
    },
    actionSheet: {
      buttonContainer: {},
      buttonText: {},
      cancelButtonContainer: {},
      cancelButtonText: {},
      titleContainer: {},
      titleText: {},
    },
    avatarWrapper: {
      container: {},
      spacer: {},
    },
    card: {
      container: {},
      cover: {},
      footer: {
        description: {},
        link: {},
        logo: {},
        title: {},
      },
    },
    container: {},
    content: {
      container: {
        borderRadiusL: 16,
        borderRadiusS: 2,
      },
      containerInner: {},
      deletedContainer: {},
      deletedText: {},
      errorContainer: {
        backgroundColor: Colors.danger,
      },
      markdown: {},
      metaContainer: {},
      metaText: {},
      textContainer: {
        borderRadiusL: 16,
        borderRadiusS: 2,
        leftBorderColor: 'rgba(0, 0, 0, 0.08)',
        leftBorderWidth: 0.5,
        rightBorderColor: 'transparent',
        rightBorderWidth: 0,
      },
    },
    file: {
      container: {},
      details: {},
      icon: {},
      size: {},
      title: {},
    },
    gallery: {
      doubleSize: 240,
      galleryContainer: {},
      halfSize: 80,
      header: {
        button: {},
        container: {},
      },
      imageContainer: {},
      single: {},
      size: 120,
      width: 240,
    },
    reactionList: {
      container: {},
      reactionCount: {},
    },
    reactionPicker: {
      column: {},
      container: {},
      containerView: {},
      emoji: {},
      reactionCount: {},
      text: {},
    },
    replies: {
      container: {},
      image: {},
      messageRepliesText: {},
    },
    status: {
      checkMark: {},
      deliveredCircle: {},
      deliveredContainer: {},
      readByContainer: {},
      readByCount: {},
      sendingContainer: {},
      sendingImage: {},
      spacer: {},
    },
  },
  messageInput: {
    actionSheet: {
      buttonContainer: {},
      buttonText: {},
      titleContainer: {},
      titleText: {},
    },
    attachButton: {},
    attachButtonIcon: {},
    container: {
      conditionalPadding: 20,
    },
    editingBoxContainer: {},
    editingBoxHeader: {},
    editingBoxHeaderTitle: {},
    fileUploadPreview: {
      attachmentContainerView: {},
      attachmentView: {},
      container: {},
      dismiss: {},
      dismissImage: {},
      filenameText: {},
      itemContainer: {},
    },
    imageUploadPreview: {
      container: {},
      dismiss: {},
      dismissImage: {},
      itemContainer: {},
      upload: {},
    },
    inputBox: {},
    inputBoxContainer: {},
    sendButton: {},
    sendButtonIcon: {},
    suggestions: {
      command: {
        args: {},
        container: {},
        description: {},
        title: {},
        top: {},
      },
      container: {
        itemHeight: 50,
        maxHeight: 250,
      },
      item: {},
      mention: {
        container: {},
        name: {},
      },
      separator: {},
      title: {},
      wrapper: {},
    },
    uploadProgressIndicator: {
      container: {},
      overlay: {},
    },
  },
  messageList: {
    dateSeparator: {
      container: {},
      date: {},
      dateText: {},
      line: {},
    },
    errorNotification: {},
    errorNotificationText: {},
    eventIndicator: {
      date: {},
      memberUpdateContainer: {},
      memberUpdateText: {},
      memberUpdateTextContainer: {},
    },
    listContainer: {},
    messageNotification: {
      container: {},
      text: {},
    },
    messageSystem: {
      container: {},
      dateText: {},
      line: {},
      text: {},
      textContainer: {},
    },
    typingIndicatorContainer: {},
  },
  spinner: {},
  thread: {
    newThread: {
      text: {},
    },
  },
  typingIndicator: {
    container: {},
    text: {
      color: Colors.textGrey,
      fontSize: 14,
    },
  },
};
