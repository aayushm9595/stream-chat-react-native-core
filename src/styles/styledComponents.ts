import * as styledComponents from 'styled-components/native';

import type { Consumer, Context } from 'react';

import type { Theme } from './themeConstants';

declare module 'styled-components/native' {
  export interface DefaultTheme extends Theme {}
}

const appStyledComponents = styledComponents as styledComponents.ReactNativeThemedStyledComponentsModule<
  Theme
>;

const {
  css,
  default: styled,
  ThemeProvider,
  useTheme,
  withTheme,
} = appStyledComponents;

const ThemeConsumer = appStyledComponents.ThemeConsumer as Consumer<Theme>;
const ThemeContext = appStyledComponents.ThemeContext as Context<Theme>;

export {
  css,
  styled,
  ThemeConsumer,
  ThemeContext,
  ThemeProvider as StyledComponentsThemeProvider,
  useTheme,
  withTheme,
};
