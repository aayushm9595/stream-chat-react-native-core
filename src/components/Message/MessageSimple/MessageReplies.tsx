import React from 'react';

import { useTranslationContext } from '../../../contexts/translationContext/TranslationContext';
import { styled } from '../../../styles/styledComponents';

import type { ImageRequireSource } from 'react-native';

import type { Message } from '../../../components/MessageList/utils/insertDates';
import type { Alignment } from '../../../contexts/messagesContext/MessagesContext';
import type {
  DefaultAttachmentType,
  DefaultChannelType,
  DefaultCommandType,
  DefaultEventType,
  DefaultMessageType,
  DefaultReactionType,
  DefaultUserType,
  UnknownType,
} from '../../../types/types';

const iconPath: ImageRequireSource = require('../../../images/icons/icon_path.png');

const Container = styled.TouchableOpacity`
  align-items: center;
  flex-direction: row;
  padding: 5px;
  ${({ theme }) => theme.message.replies.container.css}
`;

const MessageRepliesImage = styled.Image<{ alignment: Alignment }>`
  transform: ${({ alignment }) =>
    alignment === 'left' ? 'rotateY(0deg)' : 'rotateY(180deg)'};
  ${({ theme }) => theme.message.replies.image.css}
`;

const MessageRepliesText = styled.Text`
  color: ${({ theme }) => theme.colors.primary};
  font-size: 12px;
  font-weight: 700;
  ${({ theme }) => theme.message.replies.messageRepliesText.css}
`;

export type MessageRepliesProps<
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
> = {
  /**
   * Position of the message, either 'right' or 'left'
   */
  alignment: Alignment;
  /**
   * Whether or not the current message is part of a thread
   */
  isThreadList: boolean;
  /**
   * Current [message object](https://getstream.io/chat/docs/#message_format)
   */
  message: Message<At, Ch, Co, Ev, Me, Re, Us>;
  /**
   * Handler to open a thread on a message
   */
  openThread: () => void;
};

export const MessageReplies = <
  At extends UnknownType = DefaultAttachmentType,
  Ch extends UnknownType = DefaultChannelType,
  Co extends string = DefaultCommandType,
  Ev extends UnknownType = DefaultEventType,
  Me extends UnknownType = DefaultMessageType,
  Re extends UnknownType = DefaultReactionType,
  Us extends UnknownType = DefaultUserType
>(
  props: MessageRepliesProps<At, Ch, Co, Ev, Me, Re, Us>,
) => {
  const { alignment, isThreadList, message, openThread } = props;
  const { t } = useTranslationContext();
  if (isThreadList || !message.reply_count) return null;

  return (
    <Container onPress={openThread} testID='message-replies'>
      {alignment === 'left' && (
        <MessageRepliesImage
          alignment={alignment}
          source={iconPath}
          testID='message-replies-left'
        />
      )}
      <MessageRepliesText>
        {message.reply_count === 1
          ? t('1 reply')
          : t('{{ replyCount }} replies', { replyCount: message.reply_count })}
      </MessageRepliesText>
      {alignment === 'right' && (
        <MessageRepliesImage
          alignment={alignment}
          source={iconPath}
          testID='message-replies-right'
        />
      )}
    </Container>
  );
};
