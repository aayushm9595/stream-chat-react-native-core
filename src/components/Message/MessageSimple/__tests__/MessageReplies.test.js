import React from 'react';
import { ThemeProvider } from 'styled-components/native';
import {
  cleanup,
  fireEvent,
  render,
  waitFor,
} from '@testing-library/react-native';

import { generateMessage } from 'mock-builders/generator/message';
import { generateStaticUser, generateUser } from 'mock-builders/generator/user';

import { MessageReplies } from '../MessageReplies';

import { TranslationProvider } from '../../../../contexts/translationContext/TranslationContext';
import { defaultTheme } from '../../../../styles/themeConstants';

afterEach(cleanup);

describe('MessageReplies', () => {
  it('should render message replies', async () => {
    const onPressMock = jest.fn(() => null);
    const t = jest.fn((key) => key);
    const staticUser = generateStaticUser(0);
    const message = generateMessage({
      reply_count: 2,
      user: staticUser,
    });
    const {
      getByTestId,
      getByText,
      queryAllByTestId,
      rerender,
      toJSON,
    } = render(
      <TranslationProvider value={{ t }}>
        <ThemeProvider theme={defaultTheme}>
          <MessageReplies
            alignment='right'
            groupStyles={['bottom']}
            message={message}
            openThread={onPressMock}
          />
        </ThemeProvider>
      </TranslationProvider>,
    );

    await waitFor(() => {
      expect(getByTestId('message-replies')).toBeTruthy();
      expect(getByTestId('message-replies-right')).toBeTruthy();
      expect(queryAllByTestId('message-replies-left')).toHaveLength(0);
      expect(t).toHaveBeenCalledWith('{{ replyCount }} replies', {
        replyCount: message.reply_count,
      });
      expect(getByText('{{ replyCount }} replies')).toBeTruthy();
    });

    const message2 = generateMessage({
      reply_count: 1,
      user: staticUser,
    });

    rerender(
      <TranslationProvider value={{ t }}>
        <ThemeProvider theme={defaultTheme}>
          <MessageReplies
            alignment='left'
            groupStyles={['bottom']}
            message={message2}
            openThread={onPressMock}
          />
        </ThemeProvider>
      </TranslationProvider>,
    );

    fireEvent.press(getByTestId('message-replies'));
    await waitFor(() => {
      expect(onPressMock).toHaveBeenCalled();
      expect(getByTestId('message-replies')).toBeTruthy();
      expect(getByTestId('message-replies-left')).toBeTruthy();
      expect(queryAllByTestId('message-replies-right')).toHaveLength(0);
      expect(t).toHaveBeenCalledWith('1 reply');
      expect(getByText('1 reply')).toBeTruthy();
      expect(toJSON()).toMatchSnapshot();
    });
  });

  it('should not render message replies', async () => {
    const t = jest.fn((key) => key);
    const user = generateUser();
    const message = generateMessage({
      user,
    });
    const { queryAllByTestId, rerender } = render(
      <TranslationProvider value={{ t }}>
        <ThemeProvider theme={defaultTheme}>
          <MessageReplies
            alignment='right'
            groupStyles={['bottom']}
            message={message}
            openThread={() => null}
          />
        </ThemeProvider>
      </TranslationProvider>,
    );

    await waitFor(() => {
      expect(queryAllByTestId('message-replies')).toHaveLength(0);
      expect(queryAllByTestId('message-replies-left')).toHaveLength(0);
      expect(queryAllByTestId('message-replies-right')).toHaveLength(0);
    });

    const message2 = generateMessage({
      reply_count: 1,
      user,
    });

    rerender(
      <TranslationProvider value={{ t }}>
        <ThemeProvider theme={defaultTheme}>
          <MessageReplies
            alignment='right'
            groupStyles={['bottom']}
            isThreadList
            message={message2}
            openThread={() => null}
          />
        </ThemeProvider>
      </TranslationProvider>,
    );

    await waitFor(() => {
      expect(queryAllByTestId('message-replies')).toHaveLength(0);
      expect(queryAllByTestId('message-replies-left')).toHaveLength(0);
      expect(queryAllByTestId('message-replies-right')).toHaveLength(0);
    });
  });
});
