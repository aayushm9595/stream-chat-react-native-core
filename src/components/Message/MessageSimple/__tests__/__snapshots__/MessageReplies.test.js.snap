// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MessageReplies should render message replies 1`] = `
<View
  accessible={true}
  focusable={true}
  isTVSelectable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "flexDirection": "row",
      "opacity": 1,
      "paddingBottom": 5,
      "paddingLeft": 5,
      "paddingRight": 5,
      "paddingTop": 5,
    }
  }
  testID="message-replies"
>
  <Image
    alignment="left"
    source={
      Object {
        "testUri": "../../../src/images/icons/icon_path.png",
      }
    }
    style={
      Array [
        Object {
          "transform": Array [
            Object {
              "rotateY": "0deg",
            },
          ],
        },
      ]
    }
    testID="message-replies-left"
  />
  <Text
    style={
      Array [
        Object {
          "color": "#006cff",
          "fontSize": 12,
          "fontWeight": "700",
        },
      ]
    }
  >
    1 reply
  </Text>
</View>
`;
